python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -239,19 +240,42 @@ def _task_definition(self, metadata, image):
             task_definition = self.ecs.describe_task_definition(taskDefinition=self.task_definition)
             return task_definition["taskDefinition"]
 
+        secrets = merge_dicts(
+            (
+                get_tagged_secrets(self.secrets_manager, self.secrets_tag)
+                if self.secrets_tag
+                else {}
+            ),
+            get_secrets_from_arns(self.secrets_manager, self.secrets),
+        )
+        secrets_dict = (
+            {"secrets": [{"name": key, "valueFrom": value} for key, value in secrets.items()]}
+            if secrets
+            else {}
+        )
+
+        task_definition = {}
+        with suppress(ClientError):
+            # TODO: Vary family name by repository location
```

Comment:
We'll need to vary the family name (instead of it always being "dagster-run") because we don't want alternating runs launched using two different images to cause thrash and re-registration each time. I think the repository location might be the best place to source the name from?

Reply:
i like using the location - you can fetch that in a very verbose way within the run launcher like this:

```
        external_pipeline_origin = check.not_none(run.external_pipeline_origin)
        location_name = external_pipeline_origin.external_repository_origin.repository_location_origin.location_name
```

Reply:
Perfect - not the prettiest way of getting at the name, but sufficient. We'll probably need to do some basic name sanitization to make it comply with the ECS API but that's easy enough to do.